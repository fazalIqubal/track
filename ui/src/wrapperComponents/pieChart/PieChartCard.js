import React, { Component } from "react";
import Chart from "react-apexcharts";
import ReactApexChart from "react-apexcharts";

class PieChartCard extends Component {
  render() {
    return (
        <div className='pieChart'>
         <ReactApexChart id="chartOptions"
          options={
            {
              colors:  this.props.colors,
              title: {
                text: 'Tenants Statistics',
                align: 'left',
              },
              noData: {
                text: "No statistics available for tenants.",
                align: 'center',
                verticalAlign: 'middle',
                },
              chart: {
                width: 280,
                type: 'pie',
              },
              labels: this.props.labels,
              legend: { position: 'bottom', fontSize : '16px' },
              responsive: [{
                breakpoint: 280,
                options: {
                  chart: {
                    width: 200
                  },
                  legend: {
                    position: 'center'
                  }
                }
              }]
            }
          }
          series={this.props.series}
          type="pie" />
        </div>   
    );
  }
}

export default PieChartCard;