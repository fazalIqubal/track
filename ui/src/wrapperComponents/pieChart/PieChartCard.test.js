import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PieChartCard from './PieChartCard.js'
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })
import renderer from 'react-test-renderer';

const series = [44, 55, 13]
const labelsName = ['Active', 'Inactive', 'Delete']
const colors = ['Green', 'Red', 'Yellow']

describe('When Pie Chart card render', () => {
    var pieCardComponent;
    beforeEach(() => {
        pieCardComponent = shallow(<PieChartCard colors={colors} labels={labelsName} series={series}/>)
    })
    afterEach(() => {
        pieCardComponent = null;
    })
    it('renders correctly ', () => {
        const tree = renderer.create(<PieChartCard colors={colors} labels={labelsName} series={series} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should check the Div is Exist', () => {
        expect(pieCardComponent.find('.pieChart').exists()).toEqual(true)
    })
    it('should have pieChart card title text', () => {
        expect(pieCardComponent.find('#chartOptions').props().options.title.text).toEqual('Tenants Statistics')
    })
    it('should check Type of Pie Chart', () => {
        expect(pieCardComponent.find('#chartOptions').props().options.chart.type).toEqual('pie')   
    })
    it('should have Piechart color on index at 0 is Green', () => {
        expect(pieCardComponent.find('#chartOptions').props().options.colors[0]).toEqual('Green')
    })
    
    it('Should have Labels array ', () => {
        expect(pieCardComponent.find('#chartOptions').props().options.labels).toEqual(['Active', 'Inactive', 'Delete'])
    })
    it('Should have series array ', () => {
        expect(pieCardComponent.find('#chartOptions').props().series).toEqual([44, 55, 13])
    })
    it('Should have colors array ', () => {
        expect(pieCardComponent.find('#chartOptions').props().options.colors).toEqual(['Green', 'Red', 'Yellow'])
    })
    it('Should have noData text', () => {
        expect(pieCardComponent.find('#chartOptions').props().options.noData.text).toEqual("No statistics available for tenants.")
    })
})


