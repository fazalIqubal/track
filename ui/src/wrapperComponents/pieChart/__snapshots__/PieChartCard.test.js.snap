// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Pie Chart card render renders correctly  1`] = `
<div
  className="pieChart"
>
  <div
    height="auto"
    id="chartOptions"
    options={
      Object {
        "chart": Object {
          "type": "pie",
          "width": 280,
        },
        "colors": Array [
          "Green",
          "Red",
          "Yellow",
        ],
        "labels": Array [
          "Active",
          "Inactive",
          "Delete",
        ],
        "legend": Object {
          "fontSize": "16px",
          "position": "bottom",
        },
        "noData": Object {
          "align": "center",
          "text": "No statistics available for tenants.",
          "verticalAlign": "middle",
        },
        "responsive": Array [
          Object {
            "breakpoint": 280,
            "options": Object {
              "chart": Object {
                "width": 200,
              },
              "legend": Object {
                "position": "center",
              },
            },
          },
        ],
        "title": Object {
          "align": "left",
          "text": "Tenants Statistics",
        },
      }
    }
    series={
      Array [
        44,
        55,
        13,
      ]
    }
    type="pie"
    width="100%"
  />
</div>
`;
